src/Elastic.Markdown/IO/State/LinkReference.cs

```diff
@@ -15,8 +15,9 @@ public record LinkReference
 	[JsonPropertyName("url_path_prefix")]
 	public required string? UrlPathPrefix { get; init; }
 
+	/// Mapping of relative filepath and all the page's anchors for deeplinks
 	[JsonPropertyName("links")]
-	public required string[] Links { get; init; } = [];
+	public required Dictionary<string, string[]> Links { get; init; } = [];
```

Comment:
I'm wondering if an actual object instead of a string array would be more future proof.

e.g.

```json

"links": {
  "link.md": {
    "anchors": []
  }
}
```

This way we can add other attributes in the future, if necessary.

Just thinking out loud here. Also, I cannot think of anything right now.

Reply:
Good call out! we can nest maybe deprecated (renamed links) here in the future too.